src/frontend/screens/Onboarding/JoinProjectQrScreen.tsx

```diff
@@ -26,14 +27,41 @@ const m = defineMessages({
     id: "screens.Onboarding.JoinProjectQrScreen.instructionsDescription",
     defaultMessage: "Show this QR code to your Project Admin",
   },
+  instructionDescriptionAsAdmin: {
+    id: "screens.Onboarding.JoinProjectQrScreen.instructionDescriptionAsAdmin",
+    defaultMessage: "Have the Project Participant scan this invite code.",
+  },
   sendJoinRequest: {
     id: "screens.Onboarding.JoinProjectQrScreen.sendJoinRequest",
     defaultMessage: "Send Join Request instead",
   },
+  cancel: {
+    id: "screens.Onboarding.JoinProjectQrScreen.cancel",
+    defaultMessage: "Cancel",
+  },
+});
+
+//Cancels the QR code screen and resets the navigation stack so the
+//user CANNOT hit the back button to go back to this screen
+const resetAction = StackActions.reset({
+  index: 0, // <-- currect active route from actions array
+  actions: [NavigationActions.navigate({ routeName: "Home" })],
 });
 
+/* 
+  This screen is being used for 2 use cases:
+  1. Project Joiner asking to be invited to project
+  2. Project Admin adding user to project
+
+  IsAdmin:boolean Flag is passed (through navigation) if the 
+  user is in an admin
+*/
 export const JoinProjectQrScreen: NavigationStackScreenComponent = () => {
   const navigation = useNavigation();
+  const { formatMessage: t } = useIntl();
+
+  const isAdmin: boolean =
+    JSON.stringify(navigation.getParam("isAdmin", false)) === "true";
```

Comment:
do you need to stringify here if `isAdmin` is supposed to be a boolean? or will it sometimes be a string too and therefore safer to convert anyways?

Reply:
yeah.....all the documentation suggest that I stringify it....I really dont understand why, but I figure its only one line of code so why not!

Reply:
yeah makes sense. guess it's possibly to safeguard against navigation initiated by deeplinks that contain params in the link? (in which case it'd always be a string unless you parsed it)